src/compiler/checker.ts

```diff
@@ -11631,17 +11631,20 @@ namespace ts {
             //
             // When we get the type of the `Promise` symbol here, we get the type of the static
             // side of the `Promise` class, which would be `{ new <T>(...): Promise<T> }`.
-
+            
             let promiseType = getTypeFromTypeNode(node.type);
             if (promiseType === unknownType && compilerOptions.isolatedModules) {
                 // If we are compiling with isolatedModules, we may not be able to resolve the
                 // type as a value. As such, we will just return unknownType;
                 return unknownType;
             }
-
-            let promiseConstructor = getMergedSymbol(promiseType.symbol);
+            
+            let promiseConstructor = getNodeLinks(node.type).resolvedSymbol;

```

Comment:
I'm still trying to learn how the checker works, so this change is unsettling to me. Are there two almost-equivalent ways of getting the same information? 

Why can't there be just one that always does the right thing, or at least has a default that works for almost all cases?